kernelspec.sh

```diff
@@ -45,23 +49,27 @@ function install() {
     local datadir=$2
     local install_flags="--name $KERNEL_NAME"
 
-    if check_user_mode; then
-        install_flags+=" --user"
-    fi
-
-    if [[ "$install_kernel" == 'true' ]] && type jupyter >/dev/null 2>&1; then
-        jupyter kernelspec install $install_flags "$datadir"
+    if [[ "$install_kernel" == 'true' ]] && check_command jupyter; then
+        if check_user_mode; then
+            jupyter kernelspec install --user $install_flags "$datadir"
+        else
+            sudo jupyter kernelspec install $install_flags "$datadir"
```

Comment:
That should work, but maybe only on UNIX like. sudo works on Mac OS also? (no idea)

Reply:
`sudo` is fine for mac os as well, although `jupyter` in my experience is installed mostly via anaconda or locally via pip from a `brew` version of `python` in a user writable location

Reply:
If `jupyter` command is not found in `PATH`, kernelspec installation is skipped. Dou you have any good idea for finding `jupyter` command or installing kernelspec?

Reply:
Great for sudo, but indeed it's not always the best way.

@akabe if `jupyter` is not found in `PATH` I think the best could be to force the installation to fail! with a nice warning, pointing to the jupyter installation documentation as well as jupyter-ocaml documentation.

Reply:
It's troublesome to abort installation when jupyter is not found because I need to newly make `conf-jupyter` package for passing [opam-repository](https://github.com/ocaml/opam-repository)'s CI. It requires to install all dependencies by `opam depext`.
I show a jupyter-related message after opam installation: https://github.com/akabe/ocaml-jupyter/blob/master/jupyter.opam#L55-L61

Maybe I should explicitly write about manual registration of kernelspec (for users that hasn't installed jupyter yet) in README like https://github.com/akabe/ocaml-jupyter/#customize-kernel-parameters . This is the simplest idea.

Reply:
I see. There is not much that can be done, adding information in the README.md is good.